syntax = "proto3";

package inventory;

// Inventory message representing an inventory item
message Inventory_Proto {
  int32 product_id = 1;  // Foreign key to product service
  int32 quantity = 2;  // Quantity of the product in inventory
  int32 location_id = 3;  // Foreign key to Location
  string created_at = 4;  // DateTime in ISO 8601 format
  string updated_at = 5;  // DateTime in ISO 8601 format
   // RelationShip
  Location_Proto location = 6;  // One-to-one relationship: inventory is located in one location
}

// Location message representing a location
message Location_Proto {
  string location_name = 1;  // Name of the location
  string address = 2;  // Address of the location, optional
  string created_at = 3;  // DateTime in ISO 8601 format
  string updated_at = 4;  // DateTime in ISO 8601 format
   // RelationShip
  repeated Inventory_Proto inventory_items = 5;  // One-to-many relationship: a location has multiple inventory items
}
